app/controllers/partners/referrals_controller.rb

```diff
@@ -0,0 +1,38 @@
+module Partners
+    class ReferralsController < ApplicationController
+    include PromosHelper
+
+    def index 
+        respond_to do |format|
+          format.html
+          format.json  do 
+            render json: prepare_json
+          end 
+        end
+    end
+
+    private
+    def prepare_json
+        membership = Membership.find_by(user_id: current_publisher.id)
```

Comment:
I think you can just do this
```suggestion
        partner_ids = Membership.find_by(user_id: current_publisher.id).organization.map { |membership| membership.user_id}
```